fixes issue 39

valeriova96 opened the pull request:
## Fixes

### 1. updated the `create_model_from_json_schema` function in `mcpadapt/src/mcpadapt/utils/modeling.py` to ensure types like `Literal` and `Enum` are correctly preserved in the generated input schema used by the CrewAI adapter. Thus `items` and `anyOf` are added to `Field` object in `process_schema` function.

### 2. created `resolve_refs_and_remove_defs` function in `mcpadapt/src/mcpadapt/utils/modeling.py` to remove code-breaking `$defs` and `$refs` that are introduced when a custom class is used as one of the args type-hints. The function is called within `mcpadapt/src/mcpadapt/crewai_adapter.py`.

## Contributors

- @fraspecial (InSilicoTrials Technologies)
- @valeriova96 (InSilicoTrials Technologies)
- Supervised by: @marcovirgolin (InSilicoTrials Technologies)

grll commented on .DS_Store:
could you please remove this file, maybe add .DS_Store to .gitignore as well?

grll reviewed (changes requested):
Overall this look good many thanks for the PR!
What I would just add is a couple of tests for both features it brings:
1. one most simple test that would have failed under previous code for literal and enum
2. one most simple test that would have failed under previous code with nested $refs

valeriova96 commented:
Hey @grll — thanks for reviewing the PR!

I've implemented the changes you suggested. Specifically:

### 1. Added `.DS_Store` to `.gitignore`

### 2. Added two new `pytest` functions for *crewai*:

- ✅ The first test fails if `Union`, `Enum`, or a `Pydantic` class are not properly imported into `mcp_tool.inputSchema`.

- ✅ The second test triggers a `KeyError` if the script responsible for resolving `$ref`s and removing `$defs` is not integrated into the `mcp_tool.inputSchema` generation process.

grll commented on .DS_Store:
the file is still there you need to remove it as well from the git history:
```
find . -name .DS_Store -print0 | xargs -0 git rm -f --ignore-unmatch
```
and then commit / push again

valeriova96 replied:
Done 👍🏻

grll reviewed (changes requested):
looks good if you can just remove the .DS_Store from git as well then we can merge. Many thanks!

grll commented:
also if you can quickly run the linter:
`uv run ruff format .`

grll commented:
Many thanks @valerio-valente-ist and team helping on this fix!
